cate/ds/esa_cci_odp.py

```diff
@@ -1265,19 +1268,29 @@ def reporthook(block_number, read_size, total_file_size):
                     sub_monitor_msg = "file %d of %d" % (file_number, len(outdated_file_list))
                     with child_monitor.starting(sub_monitor_msg, file_size):
                         actual_url = url[_ODP_PROTOCOL_HTTP]
-                        _LOG.info(f"Downloading {actual_url} to {dataset_file}")
+
+                        if http_redirect:
+                            urlpart = actual_url.split('/', 2)[2]
+                            actual_url = f'https://{urlpart}'
+                    _LOG.info(f"Downloading {actual_url} to {dataset_file}")
+                    try:
+                        urllib.request.urlretrieve(actual_url, filename=dataset_file, reporthook=reporthook)
+                    except HTTPError as e:
                         try:
+                            actual_url = e.headers['Location']
                             urllib.request.urlretrieve(actual_url, filename=dataset_file, reporthook=reporthook)
+                            http_redirect = True
```

Comment:
How can you know `http_redirect` is now true for other URLs?

Reply:
I think, I do not quite understand: 
So the issue is, that the URL used fro request is HTTP, which does exists but is permanently redirected to HTTPS. I could let the code run though 
```
                    except HTTPError as e:
                        try:
                            actual_url = e.headers['Location']
                            urllib.request.urlretrieve(actual_url, filename=dataset_file, reporthook=reporthook)
``` 
but this would mean, that for each file the code first tries http, runs into error and then tries the url form the error header. However I do know, that the requested url belongs to the same dataset collection and therefore I thought I could assume, that if one of the files is https, then all the others would be, too.